### PR TITLE
Prevent duplicate calls to `init`

### PR DESCRIPTION
**Pull request by ThetaSinner**
### Summary

First step in addressing #3123

### TODO:
- [x] CHANGELOGs updated with appropriate info
- [x] All code changes are reflected in docs, including module-level docs

### CODE REVIEW

File: crates/holochain/src/conductor/cell.rs [hunk 1]
```diff
@@ -898,6 +898,16 @@ impl Cell {
             self.check_or_run_zome_init().await?;
         }
 
+        // The "init" function is a reserved function name that is only allowed to be called once
+        // by the `check_or_run_zome_init` function.
+        // It is acceptable to call it to get the conductor to call it once but after that we will
+        // just return a success with the same output that the init function would have returned.
+        if params.fn_name.as_ref() == "init" {
+            return Ok(Ok(ZomeCallResponse::Ok(ExternIO::encode(
```
Comment (ThetaSinner) [inline crates/holochain/src/conductor/cell.rs:906]: Commenting this out breaks the new test

**Review by c12i — APPROVED** [review]

**Review by matthme — APPROVED** [review]

File: crates/holochain/tests/tests/init.rs [hunk 2]
```diff
@@ -432,7 +433,29 @@ async fn call_init_with_invalid_parameters_from_init_across_cells() {
...
+    let callback_result = conductor
+        .call_fallible::<_, InitCallbackResult>(&zome, "init", ())
+        .await
+        .unwrap();
+
     assert!(
-        err_msg == "The callback has invalid parameters: invalid type: unit value, expected usize"
+        matches!(callback_result, InitCallbackResult::Pass),
+        "Wanted Pass but was: {:?}",
+        callback_result
```
Comment (jost-s) [inline crates/holochain/tests/tests/init.rs:459]: Call `init` manually again here and check it returns `Pass`?

Comment (ThetaSinner) [reply crates/holochain/tests/tests/init.rs:459]: Good point, I should be checking that the behavior continues

**Review by jost-s — APPROVED** [review]
Sweet

**Pull request merged**